### ISSUE TITLE
Drop baseImage configuration options

### ISSUE THREAD

**Issue by paulfantom** [coreos/prometheus-operator#3089]
**What happened?**
Started prometheus-operator and it spammed logs with:
```
msg="prometheus key=openshift-monitoring/k8s, field spec.baseImage is deprecated, 'spec.image' field should be used instead"
```

**Did you expect to see something different?**
No warning messages

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**
OpenShift 4.4

* Manifests:
https://github.com/openshift/cluster-monitoring-operator/tree/master/assets

* Prometheus Operator Logs:

```
caller=operator.go:513 component=alertmanageroperator msg="alertmanager key=openshift-monitoring/main, field spec.baseImage is deprecated, 'spec.image' field should be used instead"

caller=operator.go:1199 component=prometheusoperator msg="prometheus key=openshift-monitoring/k8s, field spec.baseImage is deprecated, 'spec.image' field should be used instead"
caller=operator.go:1209 component=prometheusoperator msg="prometheus key=openshift-monitoring/k8s, field spec.thanos.baseImage is deprecated, 'spec.thanos.image' field should be used instead"
```

**Anything else we need to know?**:
spec.baseImage config fields should be removed and usage of spec.image should be promoted.

/cc @coreos/team-monitoring

**Comment by brancz** [comment]
Removing fields are a breaking API change we can’t do that. What we can do is stop using them.

**Comment by lilic** [comment]
> Removing fields are a breaking API change we can’t do that.

I disagree, we are not 1.x, we can break the API :) These are the things we should be cleaning up before going GA.

**Comment by brancz** [comment]
The API is v1, the code handling it is not.

**Comment by lilic** [comment]
> The API is v1, the code handling it is not.

Confused why the API is v1 anywyas if code handlings is not. 🤷‍♀ 

> And kube removes entire APIs, not fields within an API.

Agreed, then lets add this to v2 API and go to v2 when we go GA, does that make sense?

**Comment by brancz** [comment]
> Agreed, then lets add this to v2 API and go to v2 when we go GA, does that make sense?

I think they're disconnected things, but makes sense.
